### PR TITLE
New package: Microsoft.ServiceFabricSDK

### PR DESCRIPTION
**Pull request by williamoconnorme**
- [x] Have you signed the [Contributor License Agreement](https://cla.opensource.microsoft.com/microsoft/winget-pkgs)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/microsoft/winget-pkgs/pulls) for the same manifest update/change?
- [x] This PR only modifies one (1) manifest
- [x] Have you [validated](https://github.com/microsoft/winget-pkgs/blob/master/AUTHORING_MANIFESTS.md#validation) your manifest locally with `winget validate --manifest <path>`?
- [x] Have you tested your manifest locally with `winget install --manifest <path>`?
- [x] Does your manifest conform to the [1.5 schema](https://github.com/microsoft/winget-pkgs/tree/master/doc/manifest/schema/1.5.0)?

Note: `<path>` is the name of the directory containing the manifest you're submitting.

---

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/microsoft/winget-pkgs/pull/125005)

### CODE REVIEW

**Comment by williamoconnorme** [comment]
@stephengillie is it possible to get more verbose logging from the pipeline installer validation?

When installing the locally, the latest SDK from Microsoft appears to install correctly once installed after the dependency package of MicrosoftServiceFabricRuntime 10.0.1949.9590 is installed.

I suspect that line 3 of my local install log below however may be why the validation is failing

```
PS C:\Users\willi\Downloads> .\MicrosoftServiceFabric.10.0.1949.9590.exe /accepteula
07/11/2023 01:52:14,Info,Installing Service Fabric Runtime. Logs written to C:\WINDOWS\TEMP\InstallFabricRuntime_638349187342157417.log
07/11/2023 01:52:14,Error,Service Fabric product installed, but cannot validate runtime & SDK compatibility due to inability to read product version from registry key 'SOFTWARE\Microsoft\Service Fabric SDK', value 'FabricSDKVersion'. Consider reinstalling.
07/11/2023 01:52:21,Info,Running extract in parallel.
07/11/2023 01:52:21,Info,Unpackaging autoextractor...
07/11/2023 01:52:21,Info,07/11/2023 01:52:21 Running Process: powershell.exe -NoProfile -Command Get-ExecutionPolicy -Scope CurrentUser with timeout 00:01:00
07/11/2023 01:52:21,Info,Creating temporary runtime directory C:\Program Files\Microsoft Service Fabric.b5i0qgt5.i3b.
07/11/2023 01:52:21,Info,Extracting runtime cab to C:\Program Files\Microsoft Service Fabric.b5i0qgt5.i3b...
07/11/2023 01:52:21,Info,07/11/2023 01:52:21 Running Process: C:\WINDOWS\TEMP\MicrosoftServiceFabricAutoextractor.exe /L "C:\Program Files\Microsoft Service Fabric.b5i0qgt5.i3b" /E /Y with timeout 01:00:00
07/11/2023 01:52:21,Info,Current Powershell Execution Policy: Unrestricted
07/11/2023 01:52:21,Info,Searching for legacy installation registry key.
07/11/2023 01:52:21,Info,Service Fabric Product not detected in registry.
07/11/2023 01:52:23,Info,Successfully extracted cab file to C:\Program Files\Microsoft Service Fabric.b5i0qgt5.i3b
07/11/2023 01:52:23,Info,Moving FabricRoot files from 'C:\Program Files\Microsoft Service Fabric.b5i0qgt5.i3b' to 'C:\Program Files\Microsoft Service Fabric'.
07/11/2023 01:52:23,Info,Moved FabricRoot files 'C:\Program Files\Microsoft Service Fabric' successfully.
07/11/2023 01:52:23,Info,Installing Service Fabric Runtime... Logs written to: C:\WINDOWS\TEMP\FabricSetupLog_638349187342157417.log
07/11/2023 01:52:23,Info,Executing powershell.exe with parameters: -NoProfile -Command & 'C:\Program Files\Microsoft Service Fabric\bin\Fabric\Fabric.Code\InstallFabric.ps1' -FabricRootIsPreInstalled -AcceptEULA -LogPath C:\WINDOWS\TEMP\FabricSetupLog_638349187342157417.log
07/11/2023 01:52:23,Info,07/11/2023 01:52:23 Running Process: powershell.exe -NoProfile -Command & 'C:\Program Files\Microsoft Service Fabric\bin\Fabric\Fabric.Code\InstallFabric.ps1' -FabricRootIsPreInstalled -AcceptEULA -LogPath C:\WINDOWS\TEMP\FabricSetupLog_638349187342157417.log with timeout 00:10:00
07/11/2023 01:52:39,Info,Successfully installed Service Fabric Runtime
```

**Comment by stephengillie** [comment]
Hi @williamoconnorme,

Yes, it's somewhat buried within the pipeline. Here, however, I believe a pipeline issue is blocking the real situation.

File: manifests/m/Microsoft/ServiceFabricSDK/7.0.1949/Microsoft.ServiceFabricSDK.installer.yaml [hunk 1]
```diff
@@ -0,0 +1,22 @@
+# Created with YamlCreate.ps1 v2.2.10 $debug=MDSU.CRLF.7-3-7.Win32NT
+# yaml-language-server: $schema=https://aka.ms/winget-manifest.installer.1.6.0.schema.json
+
+PackageIdentifier: Microsoft.ServiceFabricSDK
+PackageVersion: 7.0.1949
+InstallerType: msi
+InstallerSwitches:
+  Silent: /quiet /qn
+  Custom: /norestart
+UpgradeBehavior: install
```
Comment (stephengillie) [inline manifests/m/Microsoft/ServiceFabricSDK/7.0.1949/Microsoft.ServiceFabricSDK.installer.yaml:10]:
```suggestion
UpgradeBehavior: install
Dependencies:
  PackageDependencies:
    - PackageIdentifier: Microsoft.ServiceFabricRuntime
```

Comment (williamoconnorme) [reply manifests/m/Microsoft/ServiceFabricSDK/7.0.1949/Microsoft.ServiceFabricSDK.installer.yaml:10]: Dependencies key exists on line 18. Is this suggestion to only alter the order of the YAML keys ?

Comment (stephengillie) [reply manifests/m/Microsoft/ServiceFabricSDK/7.0.1949/Microsoft.ServiceFabricSDK.installer.yaml:10]: I'm sorry, I missed it.

**Comment by stephengillie** [comment]
> > 12:02:01 PM: Product: Microsoft Azure Service Fabric SDK -- This product requires Service Fabric Runtime to be installed.

This is due to a parsing situation with the Manual Validation pipeline - I have it find the ManifestVersion line because that's almost always the last line. Here, it's not, and below it is the dependency data. Sorry for causing concern.

**Comment by stephengillie** [comment]
image

File: manifests/m/Microsoft/ServiceFabricSDK/7.0.1949/Microsoft.ServiceFabricSDK.installer.yaml [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+Installers:
+- Architecture: x64
+  InstallerUrl: https://download.microsoft.com/download/b/8/a/b8a2fb98-0ec1-41e5-be98-9d8b5abf7856/MicrosoftServiceFabricSDK.7.0.1949.msi
+  InstallerSha256: B622775EB6D9030BA78617E2C1C6D095794E63A5600CD32A8791678163948F3E
+  ProductCode: '{A1CE7A81-5840-454F-B148-9BD65769F7BA}'
+ManifestType: installer
+ManifestVersion: 1.5.0
+Dependencies:
+  PackageDependencies:
+  - PackageIdentifier: Microsoft.ServiceFabricRuntime
```
Comment (stephengillie) [inline manifests/m/Microsoft/ServiceFabricSDK/7.0.1949/Microsoft.ServiceFabricSDK.installer.yaml:20]:
```suggestion
  - PackageIdentifier: Microsoft.VCRedist.2015+.x64
  - PackageIdentifier: Microsoft.ServiceFabricRuntime
```

**Review by stephengillie — APPROVED** [review]

**Pull request merged**